src/Tools/dotnet-trace/CommandLine/Commands/MonitorCommand.cs

```diff
@@ -0,0 +1,302 @@
+                }
+            });
+
+            while(!shouldExit.WaitOne(250))
+            {
+                while (true)
+                {
+                    if (shouldExit.WaitOne(250))
+                    {
+                        StopSession(session);
+                        return 0;
+                    }
+                    if (Console.KeyAvailable)
+                    {
+                        break;
+                    }
+                }
+                ConsoleKey cmd = Console.ReadKey(true).Key;
+                if (cmd == ConsoleKey.Q)
+                {
+                    StopSession(session);
+                    break;
+                }
+            }
+
+            return await Task.FromResult(1);
+        }
+
+        /// <summary>
+        /// Small wrapper around EventPipeSession.Stop() to catch any Timeout exceptions
+        /// A timeout exception may be thrown while stopping the session in the case of:
+        ///     1. The app exited before the user stopped the tool.
+        ///     2. The app exited after the user stopped the tool, but before the tool sent stop command.
+        /// </summary>
+        /// <param name="session"></param>
+        private static void StopSession(EventPipeSession session)
+        {
+            try
+            {
+                session.Stop();
+            }
+            catch (TimeoutException) { }
+        }
+
+        private static List<EventPipeProvider> GetWellKnownProviders(string profiles)
+        {
+            List<EventPipeProvider> providers = new List<EventPipeProvider>();
+
+            if (profiles == null)
+            {
+                return providers;
+            }
+
+            string[] profileCandidates = profiles.Split(',');
+            bool ClrProviderSet = false;
+            int ClrLevel = 0;
+            long ClrKeywords = 0;
+
+
+            foreach (string profile in profileCandidates)
+            {
+                switch (profile)
+                {
+                    case "gc-pause":
+                        ClrKeywords |= (long)ClrTraceEventParser.Keywords.GC;
+                        ClrLevel = ClrLevel < 4 ? 4 : ClrLevel;
+                        ClrProviderSet = true;
+                        break;
+                    case "loader-binder":
+                        ClrKeywords |= ((long)ClrTraceEventParser.Keywords.Binder | (long)ClrTraceEventParser.Keywords.Loader);
+                        ClrLevel = ClrLevel < 4 ? 4 : ClrLevel;
+                        ClrProviderSet = true;
+                        break;
+                    // TODO: Add more diagnostic profiles here
+                    default:
+                        throw new ArgumentException($"{profile} is not a valid diagnostic profile.");
+                }
+            }
+
+            if (ClrProviderSet)
+            {
+                providers.Add(new EventPipeProvider("Microsoft-Windows-DotNETRuntime", (EventLevel)ClrLevel, ClrKeywords));
+            }
```

Comment:
I don't think MergeProfileAndProviders is the same thing as what we're doing here. 

> I'd prefer a map of Profile --> (keyword,level) that we can index into rather than a long switch statement. 
Perhaps. Some of the profiles need to be more than just keyword,level though because there may be profiles that need more than just the runtime provider to be used. (i.e. http, TPL, etc.) - is the rationale for having this map primarily for stylistic reasons, or is there a different reason?

Reply:
> there may be profiles that need more than just the runtime provider to be used. (i.e. http, TPL, etc.)

Fair point.  The map would be more like `Profile --> IEnumerable<(provider,keyword,level)>` I guess.

It's mostly for stylistic reasons: It's easier to reason about what providers/keyowrds are in a profile if it is separated from the merging logic.  Beyond that, it would be easier to serialize/deserialize profiles if we put them in a map like that since a profile becomes a data object.  It's certainly not necessary but would help readability in my mind.

Reply:
Ok. I'll think of a way to refactor this out - thanks for the feedback!